src/main/java/org/ossgang/commons/mapbackeds/Mapbackeds.java

```diff
@@ -157,23 +245,40 @@ public M build() {
     }
 
     public static Set<Method> fieldMethods(Class<?> intfc) {
+        return collectIntfcMethods(intfc, m -> !m.isDefault());
+    }
+
+    public static Optional<Method> toStringMethod(Class<?> intfc) {
+        Set<Method> toStringMethods = toStringMethods(intfc);
+        if (toStringMethods.size() > 1) {
+            throw new IllegalArgumentException(
+                    "More than one method with a ToString annotation found in the hierarchy!");
+        }
+        return toStringMethods.stream().findFirst();
+    }
+
+    private static Set<Method> toStringMethods(Class<?> intfc) {
+        return collectIntfcMethods(intfc, m -> m.isAnnotationPresent(ToString.class));
+    }
+
+    private static Set<Method> collectIntfcMethods(Class<?> intfc, Predicate<Method> methodPredicate) {
         requireInterface(intfc);
-        Set<Method> fields = fieldsFrom(intfc);
+        Set<Method> fields = findZeroParamMethodsFrom(intfc, methodPredicate);
         Class<?>[] superinterfaces = intfc.getInterfaces();
         for (Class<?> supi : superinterfaces) {
-            fields.addAll(fieldMethods(supi));
+            fields.addAll(collectIntfcMethods(supi, methodPredicate));
         }
         return fields;
     }
 
-    private static Set<Method> fieldsFrom(Class<?> intfc) {
+    private static Set<Method> findZeroParamMethodsFrom(Class<?> intfc, Predicate<Method> methodPredicate) {
         return Arrays.stream(intfc.getDeclaredMethods()) //
                 .filter(m -> m.getParameterCount() == 0) //
-                .filter(m -> !m.isDefault()) //
+                .filter(m -> methodPredicate.test(m)) //
                 .collect(Collectors.toSet());
     }
 
-    public static <C extends Class<?>> C requireInterface(C intfc) {
+    public static <C> Class<C> requireInterface(Class<C> intfc) {
```

Comment:
private?

Reply:
Similar thatn above .. and further I use it from other classes on higher levels in another project ... so not sure ...

Reply:
So yes you are right ;-)

Reply:
I moved all the internal methods to a package private class: MapbackedInternals...

Reply:
Very good, thanks!